src/klib/kvec.h

```diff
@@ -135,6 +135,9 @@
          : 0UL)), \
      &(v).items[(i)]))
 
+#define kv_shift(v, i) ((v).size -= 1, (i) < (v).size && memmove(&kv_A(v, (i)), &kv_A(v, (i)+1), \
+                                                                 ((v).size-(i))*sizeof(kv_A(v, i))))
```

Comment:
Maybe this should be named `kv_shift1()`, or it should take an argument specifying how far to shift?